test/functional/rpc_getblockfrompeer.py

```diff
@@ -60,11 +65,30 @@ def run_test(self):
 
         self.log.info("Successful fetch")
         result = self.nodes[0].getblockfrompeer(short_tip, peer_0_peer_1_id)
-        self.wait_until(lambda: self.check_for_block(short_tip), timeout=1)
+        self.wait_until(lambda: self.check_for_block(self.nodes[0], short_tip), timeout=1)
         assert_equal(result, {})
 
         self.log.info("Don't fetch blocks we already have")
         assert_raises_rpc_error(-1, "Block already downloaded", self.nodes[0].getblockfrompeer, short_tip, peer_0_peer_1_id)
 
+        self.log.info("Connect pruned node")
+        # We need to generate more blocks to be able to prune
+        self.connect_nodes(0, 2)
+        self.generate(self.nodes[0], 400)
+        pruneheight = self.nodes[2].pruneblockchain(300)
+        assert_equal(pruneheight, 248)
+        # Ensure the block is actually pruned
+        pruned_block = self.nodes[0].getblockhash(2)
+        assert_raises_rpc_error(-1, "Block not available (pruned data)", self.nodes[2].getblock, pruned_block)
+
+        self.log.info("Fetch pruned block")
+        peers = self.nodes[2].getpeerinfo()
+        assert_equal(len(peers), 1)
+        peer_2_peer_0_id = peers[0]["id"]
+        result = self.nodes[2].getblockfrompeer(pruned_block, peer_2_peer_0_id)
+        self.wait_until(lambda: self.check_for_block(self.nodes[2], pruned_block), timeout=1)
+        assert(not "warnings" in result)
```

Comment:
I dropped the `warnings` field in 6f2133d3e6a9df23ba7922f51bdc17acd049e2e6, so this `assert` can't fail anymore...

Reply:
fixed